app/api/server/v1/users.js

```diff
@@ -933,3 +933,22 @@ API.v1.addRoute('users.listTeams', { authRequired: true }, {
 		});
 	},
 });
+
+API.v1.addRoute('users.logout', { authRequired: true }, {
+	post() {
+		const userId = this.bodyParams.userId || this.userId;
+
+		if (!hasPermission(this.userId, 'logout-other-user') && userId !== this.userId) {
```

Comment:
```suggestion
		if (userId !== this.userId && !hasPermission(this.userId, 'logout-other-user')) {
```
If we compare the IDs first, then the hasPermission method will only be called when it is necessary.

Reply:
good call